nx/lib/nx/binary_backend.ex

```diff
@@ -1233,12 +1233,18 @@ defmodule Nx.BinaryBackend do
 
   @impl true
   def cholesky(
-        %T{type: output_type, shape: {m, n}} = out,
+        %T{type: output_type, shape: output_shape} = out,
         %T{type: input_type} = tensor
       ) do
     data = to_binary(tensor)
+    rank = tuple_size(output_shape)
+    n = elem(output_shape, rank - 1)
 
-    l = B.Matrix.cholesky(data, input_type, {m, n}, output_type)
+    l =
+      bin_batch_reduce(data, n * n, input_type, <<>>, fn matrix, acc ->
+        l = B.Matrix.cholesky(matrix, input_type, {n, n}, output_type)
+        <<acc::bitstring, l::bitstring>>
```

Comment:
```suggestion
        acc <> l
```

The same applies to all similar occurences below